x-pack/solutions/observability/plugins/observability/public/components/custom_threshold/custom_threshold_rule_expression.tsx

```diff
@@ -99,6 +104,27 @@ export default function Expressions(props: Props) {
     [dataView]
   );
 
+  const fieldsQuery = useFetchAlertsRuntimeFieldsQuery(
+    {
+      http,
+      ruleTypeId: OBSERVABILITY_THRESHOLD_RULE_TYPE_ID,
+    },
+    {
+      enabled: true,
+    }
+  );
+
+  useEffect(() => {
+    const groupingFields = Object.keys(fieldsQuery.data || {});
+    if ((ruleParams.groupBy || []).length + groupingFields.length > MAX_SELECTABLE_RUNTIME_FIELDS) {
+      setGroupByError(
+        `You can't select more than ${MAX_SELECTABLE_RUNTIME_FIELDS - groupingFields.length} groups`
+      );
+    } else {
+      setGroupByError(undefined);
+    }
+  }, [fieldsQuery, ruleParams.groupBy]);
```

Comment:
This is good to prorotype so o11y can see what options they have. We'll have to note that the dropdown can contain fields that are mapped and wouldn't count towards the `MAX_SELECTABLE_RUNTIME_FIELDS` value.